### PR TITLE
Refactor Client module CleanUp

### PR DESCRIPTION
**Pull request by Darkeye14**
Fixes #Issue_Number

Please Add Screenshots If there are any UI changes.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [ ] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by Darkeye14** [comment]
@itsPronay bhai, Please review this!

File: feature/client/src/main/java/com/mifos/feature/client/syncClientDialog/SyncClientsDialogScreen.kt [hunk 1]
```diff
@@ -27,35 +38,40 @@ import androidx.compose.ui.unit.dp
...
+    viewModel: SyncClientsDialogViewModel = hiltViewModel(),
 ) {
     val uiState by viewModel.syncClientsDialogUiState.collectAsStateWithLifecycle()
     val uiData by viewModel.syncClientData.collectAsStateWithLifecycle()
 
     LaunchedEffect(key1 = Unit) {
         viewModel.syncClient()
+        list?.let {
+            viewModel.setClientList(clientsList = list)
+        }
```
Comment (itsPronay) [inline feature/client/src/main/java/com/mifos/feature/client/syncClientDialog/SyncClientsDialogScreen.kt:58]: Just a simple doubt. Shouldn't we set `clientListFirst` and `sync` later? 
image

File: feature/client/src/main/java/com/mifos/feature/client/navigation/ClientNavigation.kt [hunk 2]
```diff
@@ -59,31 +67,31 @@ fun NavGraphBuilder.clientNavGraph(
...
+            onCardClicked = { _, _ -> },
         )
         clientSurveyQuestionRoute(
-            onBackPressed = navController::popBackStack
+            onBackPressed = navController::popBackStack,
         )
         createClientRoute(
             onBackPressed = navController::popBackStack,
-            hasDatatables = { _, _ -> }
+            hasDatatables = { _, _ -> },
```
Comment (itsPronay) [inline feature/client/src/main/java/com/mifos/feature/client/navigation/ClientNavigation.kt:94]: fix missing routes here.
image

File: feature/client/src/main/java/com/mifos/feature/client/clientDetails/ui/ClientDetailsScreen.kt [hunk 3]
```diff
@@ -336,148 +341,168 @@ fun ClientDetailsScreen(
             if (showLoading) {
                 MifosCircularProgress()
             } else {
+                MifosClientDetailsScreen(
+                    padding = padding,
+                    loanAccountSelected = loanAccountSelected,
+                    savingsAccountSelected = savingsAccountSelected,
+                )
```
Comment (itsPronay) [inline feature/client/src/main/java/com/mifos/feature/client/clientDetails/ui/ClientDetailsScreen.kt:348]: good one

File: feature/client/src/main/java/com/mifos/feature/client/createNewClient/CreateNewClientScreen.kt [hunk 4]
```diff
@@ -403,109 +414,44 @@ private fun CreateNewClientContent(
...
-
-        MifosOutlinedTextField(
-            value = externalId,
-            onValueChange = { externalId = it },
-            label = stringResource(id = R.string.feature_client_external_id),
-            error = null
-        )
-
-        Spacer(modifier = Modifier.height(16.dp))
-
```
Comment (itsPronay) [inline feature/client/src/main/java/com/mifos/feature/client/createNewClient/CreateNewClientScreen.kt:508]: Is there any reason why we removing this from here?

**Review by itsPronay — CHANGES_REQUESTED** [review]

File: feature/client/src/main/java/com/mifos/feature/client/navigation/ClientNavigation.kt (hunk 2 above)
Comment (Darkeye14) [reply feature/client/src/main/java/com/mifos/feature/client/navigation/ClientNavigation.kt:94]: On it! everything has been fixed, apart from onCardClick. There seems to be an issue there.

File: feature/client/src/main/java/com/mifos/feature/client/syncClientDialog/SyncClientsDialogScreen.kt (hunk 1 above)
Comment (Darkeye14) [reply feature/client/src/main/java/com/mifos/feature/client/syncClientDialog/SyncClientsDialogScreen.kt:58]: i'll check once. It most probably is

File: feature/client/src/main/java/com/mifos/feature/client/createNewClient/CreateNewClientScreen.kt (hunk 4 above)
Comment (itsPronay) [reply feature/client/src/main/java/com/mifos/feature/client/createNewClient/CreateNewClientScreen.kt:508]: ok. did you test the funcionality? Everything works?

File: feature/client/src/main/java/com/mifos/feature/client/navigation/ClientNavigation.kt (hunk 2 above)
Comment (itsPronay) [reply feature/client/src/main/java/com/mifos/feature/client/navigation/ClientNavigation.kt:94]: about that, let's connect tonight if you are free

File: feature/client/src/main/java/com/mifos/feature/client/syncClientDialog/SyncClientsDialogScreen.kt (hunk 1 above)
Comment (itsPronay) [reply feature/client/src/main/java/com/mifos/feature/client/syncClientDialog/SyncClientsDialogScreen.kt:58]: you can take ref from `syncCenterDialogScreen` or you can check the old fragment itself

Comment (Darkeye14) [reply feature/client/src/main/java/com/mifos/feature/client/syncClientDialog/SyncClientsDialogScreen.kt:58]: Yeah, will do!!

File: feature/client/src/main/java/com/mifos/feature/client/createNewClient/CreateNewClientScreen.kt (hunk 4 above)
Comment (Darkeye14) [reply feature/client/src/main/java/com/mifos/feature/client/createNewClient/CreateNewClientScreen.kt:508]: Yes, functionality has been preserved

**Comment by Darkeye14** [comment]
https://github.com/user-attachments/assets/39171948-4ad4-4007-9ae3-2596b6693989

**Pull request merged**